cloudinit/cmd/devel/hotplug_hook.py

```diff
@@ -0,0 +1,220 @@
+# This file is part of cloud-init. See LICENSE file for license information.
```

Comment:
Some code (mainly the classes) in this file is a little more abstract than it needs to be. It's for the future support of disk events. Given that Ryan initially wrote it that way and it works and makes sense for future additions, it didn't make sense to make it any simpler.